components/camera/transformpipeline/transform.go

```diff
@@ -32,26 +32,32 @@ const (
 	transformTypeDepthPreprocess = transformType("depth_preprocess")
 )
 
-// emptyConfig is for transforms that have no attribute fields.
-type emptyConfig struct{}
-
 // transformRegistration holds pertinent information regarding the available transforms.
 type transformRegistration struct {
 	name        string
 	retType     interface{}
 	description string
 }
 
+// Do not share the share the same config type across multile configs because otherwise
+// multiple json-schema reflected types will share the same $id
+// (which is not valid jsonschema and will fail to compile on the FE).
+type (
+	idenityConfig         struct{}
+	depthPrettyConfig     struct{}
+	depthPreprocessConfig struct{}
+)
+
 // registeredTransformConfigs is a map of all available transform configs, used for populating fields in the front-end.
 var registeredTransformConfigs = map[transformType]*transformRegistration{
 	transformTypeIdentity: {
 		string(transformTypeIdentity),
-		&emptyConfig{},
+		&idenityConfig{},
 		"Does nothing to the image. Can use this to duplicate camera sources, or change the source's stream or parameters.",
 	},
 	transformTypeRotate: {
 		string(transformTypeRotate),
-		&emptyConfig{},
+		&rotateConfig{},
```

Comment:
is there a reason this had `emptyConfig` before instead of `rotateConfig`? Bug?

Reply:
probably a bug! Thanks for catching it